New template for search bar, with styles and placeholder text as parameters

danielfmiranda opened the pull request:
Closes #8778 
Related PRs/issues #



Instead of defining a new template for the blog search bar with nearly identical code as the research hub search bar, I instead created a new template just titled "search_bar.html", where I can then pass in parameters for things such as the icon styling, input styling, and also placeholder text. 

Screenshots:
---

Research Hub using new template, and passing styles and placeholder text as params:

<img width="1488" alt="Screen Shot 2022-06-09 at 4 39 22 PM" src="https://user-images.githubusercontent.com/18314510/172964352-b17f37c8-00f2-4228-9837-7f8cc3f7cd87.png">


Blog Index Page using new template, and passing styles and placeholder text as params (not exactly following Figma, this was just meant to show that we can use the template and pass in different styles as needed):

<img width="1488" alt="Screen Shot 2022-06-09 at 4 39 50 PM" src="https://user-images.githubusercontent.com/18314510/172964434-b12f9336-ef4e-4266-8338-ce55d73b4408.png">

danielfmiranda commented on network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/search_bar.html:
```diff
@@ -0,0 +1,8 @@
+{% load i18n %}
+<div class="tw-relative">
+  <label for="search" class="tw-sr-only">{% translate 'Search' context 'Button' %}</label>
+  <input type="text" value="{{ search_query }}" name="search" id="search" placeholder="{% translate placeholder_text context 'Search input' %}" class="tw-w-full form-control tw-h-auto tw-p-4 tw-pr-7 tw-rounded-none {{input_styles}}" />
```
Hi @TheoChevalier! I was wondering if you could take a look and let me know if this is proper use of the translate block, as I am passing in the text as a parameter from "research_search_bar" and "blog_search_bar".

Thanks in advance!

tbrlpld replied:
@danielfmiranda I would be surprised if this works, because you can pass any text dynamically to the template. I think you will need to move the translation into the template where the text is actually defined. There you can use `{% translate "My new placeholder text" context "Search input placeholder text" as placeholder_text %}` and then pass the `placeholder_text` variable to the included template, like so `{% include "wagtailpages/fragments/search_bar.html" with placeholder_text=placeholder_text input_styles="tw-border-gray-40" %}`

danielfmiranda commented on network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/research_search_bar.html:
```diff
@@ -1,8 +1 @@
-    <svg fill="none" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg" stroke="currentColor"><g stroke-linecap="round" stroke-linejoin="round" stroke-width="2"><circle cx="-14.5" cy="9" r="8" transform="scale(-1 1)"/><path d="m8.682 14.818-7.682 7.682" transform="matrix(-1 0 0 -1 9.682 37.318)"/></g></svg>
-  </button>
-</div>
+{% include "wagtailpages/fragments/search_bar.html" with placeholder_text="Search our research..." input_styles="tw-border-gray-40" %}
```
The reason for this new template is so we do not have to update classes on every template where the research hub search bar is being used, if there is a future design change

tbrlpld reviewed (changes requested):
@danielfmiranda I think this is going great. I would suggest to make the blog styles the default ones, because that seems to be more in line with the rest of the Mozilla design language. Override what you need for the research hub variant. And I would honestly say, just use the icon direction from the blog design and just keep that for the research hub too. Makes it more consistent.

danielfmiranda commented on network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/search_bar.html:
```diff
@@ -0,0 +1,8 @@
+{% load i18n %}
+<div class="tw-relative">
+  <label for="search" class="tw-sr-only">{% translate 'Search' context 'Button' %}</label>
+  <input type="text" value="{{ search_query }}" name="search" id="search" placeholder="{% translate placeholder_text context 'Search input' %}" class="tw-w-full form-control tw-h-auto tw-p-4 tw-pr-7 tw-rounded-none {{input_styles}}" />
```
Great Idea, I have added this!

danielfmiranda commented:
Hi @tbrlpld!

Thank you for the great suggestions, and for the tip on how to get translated strings working with the search bar.

I have implemented the request to pass the translated strings as a parameter, and also updated the direction of the icon so both templates match. I also updated the master search bar template to have the blog styling as a default, but users can still add custom styles as a parameter.

Thanks again!
